src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Win32.cs

```diff
@@ -869,41 +760,42 @@ private static string GetLocalizedNameByMuiNativeResource(string resource)
                 return string.Empty;
             }
 
-            string filePath;
+            // Get the resource ID
+            if (!int.TryParse(resources[1], NumberStyles.Integer, CultureInfo.InvariantCulture, out int resourceId))
+            {
+                return string.Empty;
+            }
+            resourceId = -resourceId;
+
+            // Use the current UI culture when culture not specified
+            CultureInfo cultureInfo = CultureInfo.CurrentUICulture;
 
             // get the path to Windows\System32
             string system32 = Environment.SystemDirectory;
 
-            // trim the string "@tzres.dll" => "tzres.dll"
-            ReadOnlySpan<char> tzresDll = resources[0].AsSpan().TrimStart('@');
+            // trim the string "@tzres.dll" to "tzres.dll" and append the "mui" file extension to it.
+            string tzresDll = $"{resources[0].AsSpan().TrimStart('@')}.mui";
 
             try
             {
-                filePath = Path.Join(system32, tzresDll);
+                while (cultureInfo.Name.Length != 0)
+                {
+                    string filePath = Path.Join(system32, cultureInfo.Name, tzresDll);
+                    if (File.Exists(filePath))
+                    {
+                        // Finally, get the resource from the resource path
+                        return GetLocalizedNameByNativeResource(filePath, resourceId);
+                    }
+
+                    cultureInfo = cultureInfo.Parent;
+                }
             }
             catch (ArgumentException)
             {
                 // there were probably illegal characters in the path
-                return string.Empty;
-            }
-
-            // Get the MUI File Path
-            string fileMuiPath = GetCachedFileMuiPath(filePath, cultureInfo);
-            if (fileMuiPath == string.Empty)
-            {
-                // not likely, but we could not resolve a MUI path
-                return string.Empty;
             }
 
-            // Get the resource ID
-            if (!int.TryParse(resources[1], NumberStyles.Integer, CultureInfo.InvariantCulture, out int resourceId))
-            {
-                return string.Empty;
-            }
-            resourceId = -resourceId;
-
-            // Finally, get the resource from the resource path
-            return GetLocalizedNameByNativeResource(fileMuiPath, resourceId);
+            return string.Empty;
```

Comment:
If I'm following correctly, if the requested culture or its parent is not found, then the string returns empty? I think this will be a problem.  For example, on my system with `en-US` language installed, there are folders under `C:\Windows\System32` for `en`, `en-au`, `en-GB`, `en-sg`, and `en-US`.  However, the `tzres.dll.mui` file is *only* found under `en-US`.  Thus if I set the .NET `CurrentUICulture` to `en-GB`, it won't find either `C:\Windows\System32\en-GB\tzres.dll.mui` or `C:\Windows\System32\en\tzres.dll.mui`, and thus I will get an empty string instead of the English time zone name.

I believe the algorithm needs to search:
- The specific culture first
- Then its parent
- Then any children of the same parent
- Then fallback to the default UI culture (not ever returning empty string)

Reply:
We discussed that on the email, and I explained how resource lookup work in .NET. .NET always lookup the parent chain only and then fallback to neutral resources. When returning empty string it doesn't mean we'll return it as the display name, it means we'll just fallback to the default display name on the system. this is like what we were doing before.

Reply:
The private `GetFileMuiPath` function in `TimeZoneInfo.Win32.cs` that I added does indeed use steps 3 and 4 - searching for a child of the same parent.  Before that, we weren't searching at all, but were just using the default OS UI language.

Reply:
Basically the problem comes when you've got say `en-US` as default and `es-ES` installed also.  If the user asks for a time zone string for `es-MX`, we need to return the one from `es-ES`, not the one from `en-US`.

Reply:
I believe returning the default from the system is better than OS UI one. We'll be consistent across all configurations. This is more predictable. This matches the resource lookup in general too to always fallback to specific set of resources.
Also, I explained a concerns in my email communication regarding step 3 which I believe it is wrong to do it.

Reply:
Ok.  I can understand your point of view.  Let's keep it as you suggest.  Thanks.

Reply:
I agree with @tarekgh — we should not invent a new resolution mechanism just for time zones.